src/common/dump.cpp

```diff
@@ -230,3 +230,54 @@ int dump_hardfault_test_1(void) {
     volatile int c = 1 / b;
     return c;
 }
+
+// Dumping error message
+
+void dump_err_to_xflash(const char *error, const char *title) {
+    // This is called after w25x initialization (in dumpinfo_t dumping), so no additional w25x_init() is needed
+    uint8_t valid_flag = dump_err_in_xflash_get_valid_flag();
+    if (valid_flag != 0xFF) {                                       // ==> message space is not empty
+        if (valid_flag != 0 || dump_err_in_xflash_is_displayed()) { // ==> erase sector if flag is corrupted (!= 0x00) or is valid and is displayed
+            w25x_sector_erase(w25x_error_start_adress);
+            w25x_fetch_error();
+        }
+    }
+    dumpmessage_t dump_message;
+    dump_message.not_displayed = dump_message.invalid = 0;
+    strlcpy(dump_message.title, title, sizeof(dump_message.title));
+    strlcpy(dump_message.msg, error, sizeof(dump_message.msg));
+    w25x_program(w25x_error_start_adress, (uint8_t *)(&dump_message), sizeof(dumpmessage_t));
```

Comment:
Here you write whole dump_message. So any random data contained in `not_displayed` are written also. Even if you would initialize `not_displayed` field into `0xff` still requirement not to write the same memory location would be violated.

I would suggest to split w25x_program() into two or more calls. Write invalid flag, write messages, do not write anything else.

Reply:
It was initialized to 0, which is wrong of course, here is a fix `9d28bd0`